agent/src/IngestDetailsRepository.php

```diff
@@ -98,7 +100,12 @@ private function refresh(): PromiseInterface
 
                 call_user_func($this->onAuthenticationSuccess, $ingestDetails, $duration);
 
-                $this->quotaExceeded = false;
+                if ($this->overQuota) {
+                    $this->overQuota = false;
+
+                    call_user_func($this->onUnderQuota);
+                }
```

Comment:
Just a thought on the naming. I wonder if it makes sense to name those something like below. A successful authentication doesn't mean the quota has been restored, but it means that we should attempt to resume ingestion.

```suggestion
                if ($this->ingestionPaused) {
                    $this->ingestionPaused = false;

                    call_user_func($this->resumeIngestionHandler);
                }
```

Reply:
I prefer the current naming. The `IngestDetailsRepository` isn't concerned with the process of ingesting records, so I don't want to give it knowledge of that process or the state of the process.

I think this just feels weird because right now a successful authentication is our mechanism for being marked under quota. That will change once the app lets us know if we continue to be under quota, at which point this code will change to be something like:

```php
if ($this->overQuota !== $details->overQuota) {
    $this->overQuota = $details->overQuota;

    $this->overQuota
        ? call_user_func($this->onOverQuota)
        : call_user_func($this->onUnderQuota);
}
```

Reply:
That makes a lot of sense, especially when the app lets us know about the quota!